test/rocarray/random.jl

```diff
@@ -0,0 +1,66 @@
+# NOTE: tests should cover both pow2 and non-pow2 dims
+
+# in-place
+for (f,T) in ((rand!,Float32),
+              (rand!,Cuint),
+              (randn!,Float32),
+              (rand_logn!,Float32),
+              (rand_poisson!,Cuint)),
+    d in (2, (2,2), (2,2,2), 3, (3,3), (3,3,3))
+    A = ROCArray{T}(undef, d)
```

Comment:
Maybe we can initialize `A` with zeros, and then test that the array is non-zero after `f(A)`?